packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -62,7 +74,8 @@ export class ApplicationExtension<
      * @param routes - All the application routes from both extensions and base application.
      * @returns routes - The modified application routes.
      */
-    public beforeRouteMatch(routes: RouteProps[]): RouteProps[] {
-        return routes
+    public beforeRouteMatch(params: BeforeRouteMatchParams): RouteProps[] {
```

Comment:
I like the use of the params argument and it's XXParams naming, should work well when we need to add additional arguments if it comes down to that in the future 👍